src/build_system.ml

```diff
@@ -394,81 +400,84 @@ let set_rule_generators ~init ~gen_rules =
   Fdecl.set t.init_rules init_rules;
   Fdecl.set t.gen_rules gen_rules
 
-let get_dir_status t ~dir =
-  match Path.Table.find t.dirs dir with
-  | Some x -> Some x
+let get_dir_triage t ~dir =
+  match Path.as_in_source_tree dir with
+  | Some dir ->
+    Dir_triage.Known (Non_build (
+      Path.Source.Set.to_list (File_tree.files_of t.file_tree dir)
+      |> List.map ~f:Path.source
+      |> Path.Set.of_list))
   | None ->
-    let res =
-      match Path.as_in_source_tree dir with
-      | Some dir ->
-        Some (Dir_status.Loaded {
+    if Path.equal dir Path.build_dir then
+      (* Not allowed to look here *)
+      Dir_triage.Known
+        (Build
+           { rules_produced = Rules.empty;
```

Comment:
I see this value being produced in a couple of places. It would be good to have a `Loaded.no_rules` or `Loaded.empty_build` value instead.

Reply:
Chose `no_rules`.